mirror-godot-app/script/gd/gdscript_instance.gd

```diff
@@ -295,6 +328,36 @@ func _sync_gdscript_code_with_entries() -> void:
 		_source_code = entry.sync_gdscript_code_with_entry(_source_code)
 
 
+func _sync_exposed_variables_with_spaceobj_spacevars() -> void:
+	#_update_exposed_variables()
+	for var_name in _exposed_var_names:
+		if Mirror.has_object_variable(target_node, var_name):
+			script_instance_object.set(var_name, Mirror.get_object_variable(target_node, var_name))
+		else:
+			Mirror.set_object_variable(target_node, var_name, script_instance_object.get(var_name))
+
+
+func _on_load_exposed_vars() -> void:
+	for var_name in _exposed_var_names:
+		script_instance_object.set(var_name, Mirror.get_object_variable(target_node, var_name))
+
+
+func _on_save_exposed_vars() -> void:
+	for var_name in _exposed_var_names:
+		Mirror.set_object_variable(target_node, var_name, script_instance_object.get(var_name))
```

Comment:
> thing = myvalue followed by save_exposed_vars.emit() (plus there is also load_exposed_vars.emit() if you want to read the space vars back into the script vars, this might be useful for sharing data between scripts).
What if we have the script automatically call `save_exposed_vars.emit()` after an assignment like `thing = myvalue`? I feel like users will often forget to call that and be confused. Or is there a technical reason that wouldn't work?

Reply:
On the technical side, I don't know how to detect when that happens or inject this code when that happens.

Reply:
Fair point, we can merge it for now and optimize later